lib/model/CDetectionRule.cc

```diff
@@ -14,30 +12,20 @@
 namespace ml {
 namespace model {
 
-CDetectionRule::CDetectionRule()
-    : m_Action(E_FilterResults), m_Conditions(), m_ConditionsConnective(E_Or),
-      m_TargetFieldName(), m_TargetFieldValue() {
-    m_Conditions.reserve(1);
-}
-
 void CDetectionRule::action(int action) {
     m_Action = action;
 }
 
-void CDetectionRule::conditionsConnective(EConditionsConnective connective) {
-    m_ConditionsConnective = connective;
-}
-
-void CDetectionRule::addCondition(const CRuleCondition& condition) {
-    m_Conditions.push_back(condition);
+void CDetectionRule::includeScope(std::string field, const core::CPatternSet& filter) {
+    m_Scope.include(field, filter);
```

Comment:
Use `std::move(field)`, as it was passed by value and is not used anywhere else.

Same in `excludeScope` immediately below.